libbeat/common/version.go

```diff
@@ -150,3 +151,25 @@ func (v *Version) metaIsLessThanOrEqual(v1 *Version) bool {
 	}
 	return v.Meta <= v1.Meta
 }
+
+// UnmarshalJSON unmarshals a JSON string version representation into a Version struct
+// Implements https://golang.org/pkg/encoding/json/#Unmarshaler
+func (v *Version) UnmarshalJSON(version []byte) error {
+	var versionStr string
+	err := json.Unmarshal(version, &versionStr)
+	if err != nil {
+		return err
+	}
+
+	ver, err := NewVersion(versionStr)
+	if err != nil {
+		return err
+	}
+
+	if ver == nil {
+		return fmt.Errorf("could not unmarshal version from JSON")
+	}
```

Comment:
Ah right, I misunderstood your previous comment:

>  If NewVersion fails can return a nil ver?

But yeah, since we try to dereference `ver` right after, I think checking that it's not `nil` is okay.